packages/lodestar/src/opPool/opPool.ts

```diff
@@ -74,4 +76,39 @@ export class OpPool extends EventEmitter {
     ];
     await Promise.all(tasks);
   }
+
+  public async checkDuplicateProposer(config, block: BeaconBlock): Promise<void> {
+    const epoch: Epoch = computeEpochOfSlot(config, block.slot);
+    const proposers: Map<ValidatorIndex, Slot> = this.proposers.get(epoch);
+    const proposerIndex: ValidatorIndex = await this.db.getValidatorIndex(block.signature);
+
+    // Check if proposer already exists
+    if (proposers.get(proposerIndex)) {
+      const existingSlot: Slot = this.proposers.get(epoch).get(proposerIndex);
+      const prevBlock: BeaconBlock = await this.db.block.getBlockBySlot(existingSlot);
+
+      // Create slashing
+      const slashing: ProposerSlashing = {
+        proposerIndex: proposerIndex,
+        header1: {
+          stateRoot: prevBlock.stateRoot,
+          signature: prevBlock.signature,
+          slot: prevBlock.slot,
+          parentRoot: prevBlock.parentRoot,
+          bodyRoot: hashTreeRoot(prevBlock.body, config.types.BeaconBlockBody),
+        },
+        header2: {
+          stateRoot: block.stateRoot,
+          signature: block.signature,
+          slot: block.slot,
+          parentRoot: block.parentRoot,
+          bodyRoot: hashTreeRoot(block.body, config.types.BeaconBlockBody),
+        }
+      };
+      this.db.proposerSlashing.set(proposerIndex, slashing);
```

Comment:
`this.proposerSlashing.receive(slashing)`

Reply:
receive doesn't exist though?

Reply:
it most certainly does. :)

https://github.com/ChainSafe/lodestar/blob/master/packages/lodestar/src/opPool/modules/abstract.ts#L11

Reply:
weiiird my IDE is being a potatoe